pkg/util/fs_util.go

```diff
@@ -55,6 +62,10 @@ func GetFSFromImage(root string, img v1.Image) error {
 
 	fs := map[string]struct{}{}
 	whiteouts := map[string]struct{}{}
+	hardlinks, err := retrieveHardlinks(layers)
```

Comment:
Do you think it would make sense to process each layer separately instead of doing them all at once? I think we already disallow hardlinks to cross layer boundaries.

Reply:
Yah I think that would be better, I changed it to retrieve hardlinks as we process layers